src/Stryker.Core/Stryker.Core/Mutators/NegateConditionMutator.cs

```diff
@@ -44,8 +61,18 @@ public override IEnumerable<Mutation> ApplyMutations(ExpressionSyntax node)
             }
         }
 
-        private static PrefixUnaryExpressionSyntax NegateCondition(ExpressionSyntax expressionSyntax)
+        private static ExpressionSyntax NegateCondition(ExpressionSyntax expressionSyntax)
         {
+            if (expressionSyntax.Kind() == SyntaxKind.LogicalNotExpression && expressionSyntax is PrefixUnaryExpressionSyntax prefixUnary)
+            {
+                return prefixUnary.Operand;
+            }
+
+            if (expressionSyntax is BinaryExpressionSyntax binaryExpression && _converters.ContainsKey(expressionSyntax.Kind()))
+            {
+                return SyntaxFactory.BinaryExpression(_converters[expressionSyntax.Kind()], binaryExpression.Left,
```

Comment:
No worries, it's all good.

The biggest issue is: We're not sure what to do. We're gonna talk about it internally and take your points into consideration as well. I think this is also the last thing we need to think for this PR as a whole, it can probably be merged once we resolve this issue. Sorry to keep you waiting a bit.

Reply:
We've talked and think the negate mutator should not be placing binary mutations. This is for the sake of simplicity. @dupdob could you remove the binary mutation from this mutator?

Reply:
No problem undoing this. I guess duplicate mutation detection has little value then.
As a contributor, I don't mind.
As a user, I can't say I am happy with that decision.

Reply:
Why do you say it has little value? This for sure is a nice improvement on our current behaviour. It's just that we don't want to get the mutators involved with each other. Also we want to keep the mutators in sync with the other strykers. So it's not our decision only. I'm sorry that you're not happy with this... 

Lets discuss this further in the issue indeed.

Reply:
Regarding duplicate mutants: if mutators build very specific constructs, it is very unlikely we end up having duplicates at the syntax tree level; but we may still have semantic duplicates.
eg, as of today
`if (a || b)` will be mutated to:
- `if (a && b)`
- `if (!(a || b))`
note the same syntax, but the same semantic result

Btw, I just realised I am the one who introduced the filtering in NegateCondition so that it does not mutate construction that will be mutated by the BinaryExpressionMutator.

Reply:
One solution could be to improve the duplicate detector so that it can detect some/known equivalent mutations (i.e. ! ( == ) vs ( != )), but this leads to more complex code

Reply:
Hi! Perhaps it was a bit unclear. We decided that we would like the Negate Condition Mutator to not create any binary expression mutation. We do like that you fixed the issue of duplicate mutations!

Reply:
Ok. I suggest we move this conversation to issue #1170.
To put it short: the filtering is inadequate as it ends up removing none duplicate as of now and it was supposed to be a stop gap until the binary expression mutation was put in place (to be removed once done).

Due to the impact of this refactoring, I propose this PR to be integrated as is (i.e. without any duplicate mutant management) and discuss